src/app/find-the-expert/page.tsx

```diff
@@ -48,13 +50,27 @@ const FindTheExpertPage = async () => {
         </span>
         <span className="block sm:inline"> Tech Survey - Find the expert</span>
       </h1>
-      {session ? <ContentSection /> : <LoginSection session={session} />}
+      {session ? <ContentSection role={context.searchParams.role} tech={context.searchParams.tech}/> : <LoginSection session={session} />}
     </div>
   );
 };
 
-const ShowTableWrapper = async () => {
-  const userAnswersForRole = await fetchUserAnswersForRole();
+async function getUserAnswers(tech:string, role:string) {
+  switch (true) {
+    case (tech == undefined && role != undefined):
+      return await fetchUserAnswersForRole(role);
+    case (tech != undefined && role == undefined):
+      return await fetchUserAnswersForQuestion(tech);
+    case (tech != undefined && role != undefined):
+      return await fetchUserAnswersForRoleAndQuestion(role, tech);
+    default:
+      return await fetchUserAnswers();
+  }
+}
```

Comment:
You can use the `falsy` and `truthy` of JavaScript to achieve the same thing here in a much simpler way:

```js
async function getUserAnswers(tech?: string, role?: string) {
  if (!tech && role) return fetchUserAnswersForRole(role);
  if (tech && !role) return fetchUserAnswersForQuestion(tech);
  if (tech && role) return fetchUserAnswersForRoleAndQuestion(role, tech);
  
  return fetchUserAnswers();
}
```